src/server.h

```diff
@@ -1765,7 +1765,9 @@ struct redisServer {
     char replid2[CONFIG_RUN_ID_SIZE+1]; /* replid inherited from master*/
     long long master_repl_offset;   /* My current replication offset */
     long long second_replid_offset; /* Accept offsets up to this for replid2. */
-    redisAtomic long long pot_fsynced_reploff;/* Largest replication offset to _potentially_ have been fsynced */
+    redisAtomic long long pot_fsynced_reploff;/* Largest replication offset to _potentially_ have been fsynced. */
+                                              /* If no replication is configured, this will track a simple */
```

Comment:
maybe instead of `if no replication is configured` you should say `when replication offset isn't being tracked`?

it's a bit ugly that this same variable tracks a completely different thing on different states, we can resolve it by either changing it to track unsynced bytes somehow, or if we want to keep tracking an epoch, maybe we should have two separate variables?
the variable that tracks epochs can always be incremented, and just ignored depending on the state.
i'm in favor of separate variables, the code that updates them will be rather straight forward, and the code that uses them a bit more explicit (so that's the one that'll require comments).